enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,510 @@
+}
+
+type Region struct {
+  // The name is the region name to be created as a tag
+  // in vSphere and attached to a datacenter and used
+  // with the cloud provider config
+  // +required
+  Name string `json:"name,omitempty"`
+
+  // Datacenter is the name of the datacenter to use in the vCenter.
+  // +required
+  Datacenter string `json:"datacenter,omitempty"`
+
+  // +required
+  Zones []Zone `json:"zones,omitempty"`
+}
+
+type Zone struct {
+  // Name is the zone name to be created as a tag in vSphere
+  // and attached to a cluster and used with the cloud provider config
+  // +required
+  Name string `json:"name,omitempty"`
+  // +required
+  Cluster string `json:"cluster,omitempty"`
+  // Network specifies the name of the network to be used by the cluster.
+  // +required
+  Network string `json:"network,omitempty"`
+  // ResourcePool is the absolute path of the resource pool where virtual machines will be
+  // created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+  // +optional
+  ResourcePool string `json:"resourcepool,omitempty"`
+  // Datastore is the default datastore to use for provisioning volumes.
+  // +required
+  Datastore string `json:"datastore,omitempty"`
+}
+```
+
+Add [platform validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/platform.go)
+for the new struct fields that are required.
+
+#### MachinePool
+
+The `MachinePool` [struct](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/machinepool.go#L5-L26)
+needs a single change to include the zones.
+The `Zones` like the other cloud providers will determine the location
+where the virtual machine will reside within a vSphere environment.
+
+```golang
+type MachinePool struct {
+...
+  Zones []string `json:"zones,omitempty"`
+}
+```
+
+Add [machinepool validation](https://github.com/openshift/installer/blob/master/pkg/types/vsphere/validation/machinepool.go)
+the zone name has a corresponding entry in `Regions.Zones`.
+
+#### Cloud Config
+
+The out-of-tree CCM is required for this change. The out-of-tree CCM also updates
+the cloud-config ini configuration. Below is an example of what the
+cloud-config will change to. A `VirtualCenter` section will be added per
+vcenter in `vcenters`. `datacenters` is a comma-delimited string that will contain
+all the datacenters per region.
+
+```ini
+[Global]
+  insecure-flag = "1"
+  secret-name = "vsphere-creds"
+  secret-namespace = "kube-system"
+
+[VirtualCenter "10.0.0.1"]
+  datacenters = "SDDC-Datacenter"
+
+[Labels]
+  region = openshift-region
+  zone = openshift-zone
+```
+
+There is an outstanding question if the out-of-tree cloud provider will be
+enabled by default. If this does not happen of 4.12 we will need to include
+the appropriate configurations. It will be more convenient to switch to the
+yaml-based configuration of the provider and leave the existing in-tree
+ini-based cloud-config alone.
+
+#### TUI
```

Comment:
What's TUI?

Reply:
The installer cli terminal user interface